views/components/ItemList.jsx

```diff
@@ -22,7 +22,7 @@ class ItemList extends React.Component {
       status: 'pending'
     };
 
-    this.props.addItem(newItem);
+    if (!!newItem.text.trim()) this.props.addItem(newItem);
```

Comment:
Really interesting way to check for blank text. I think the lines below should also be included in this conditional, though.